conan/tools/microsoft/visual.py

```diff
@@ -275,3 +276,39 @@ def is_msvc_static_runtime(conanfile):
     :return: True, if msvc + runtime MT. Otherwise, False
     """
     return is_msvc(conanfile) and "MT" in msvc_runtime_flag(conanfile)
+
+
+def msvs_toolset(conanfile):
+    """ Returns the corresponding Visual Studio/msvc platform toolset based on the settings of the
+        given conanfile
+    :param conanfile: Conanfile instance
+    :return: Microsoft toolset when compiler is valid. Otherwise, an empty string.
+    """
+    settings = conanfile.settings
+    compiler = settings.get_safe("compiler")
+    compiler_version = settings.get_safe("compiler.version")
+    if compiler == "msvc":
+        subs_toolset = settings.get_safe("compiler.toolset")
+        if subs_toolset:
+            return subs_toolset
+        return msvc_version_to_toolset_version(compiler_version)
+    if compiler == "intel":
+        compiler_version = compiler_version if "." in compiler_version else \
```

Comment:
may raise if `compiler_version` is `None`

Reply:
good catch, instead, I'll check `compiler == intel and compiler_version`